src/integration_tests/action_transition_multicopter_fixedwing.cpp

```diff
@@ -44,11 +33,35 @@ void takeoff_and_transition_to_fixedwing()
     auto action = std::make_shared<Action>(system);
     auto telemetry = std::make_shared<Telemetry>(system);
 
+    while (!telemetry->health_all_ok()) {
+        std::cout << "waiting for system to be ready" << '\n';
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
     // We need to takeoff first, otherwise we can't actually transition
-    takeoff(action, telemetry);
+    float altitude_m = 10.0f;
+    action->set_takeoff_altitude(altitude_m);
+
+    Action::Result action_ret = action->arm();
+    ASSERT_EQ(action_ret, Action::Result::Success);
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    LogInfo() << "Taking off";
+    action_ret = action->takeoff();
+    ASSERT_EQ(action_ret, Action::Result::Success);
 
     LogInfo() << "Waiting until hovering";
-    wait_until_hovering(telemetry);
+    auto prom = std::promise<void>{};
+    auto fut = prom.get_future();
+    // Wait until hovering.
+    telemetry->subscribe_flight_mode([&telemetry, &prom](Telemetry::FlightMode mode) {
+        if (mode == Telemetry::FlightMode::Hold) {
+            telemetry->subscribe_flight_mode(nullllptr);
```

Comment:
`nullllptr`? :sweat_smile:

Reply:
Haha, that must have slipped in after testing.

Reply:
Ah yes, I remember typing `all` for `git add --all` and it was in the wrong window but I couldn't see where it went.

Reply:
Also, beware of `nullllptr`, they are extra null, at least double!